src/Sentry/Platforms/Cocoa/SentrySdk.cs

```diff
@@ -239,6 +234,11 @@ private static CocoaSdk.SentryHttpStatusCodeRange[] GetFailedRequestStatusCodes(
             }
         }
 
+        if (hub is DisabledHub)
```

Comment:
Maybe worth referencing or summarising https://github.com/getsentry/sentry-dotnet/pull/4521#discussion_r2347616896 in the code here to explain why this happens after the suppression code above.

As an aside, this is a bit of a gotcha for custom event processes that users implement. They may not to work for native exceptions that crash the process, since events for those will likely get fired off before the .NET SDK has been initialised, the next time the process runs.